From 1a9ddf907792de0778f13248b6f0bd7e8d211562 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Thu, 16 Mar 2017 15:44:48 -0400
Subject: [PATCH] Try to fit in a sandbox

When running in another apps sandbox, use an app name that works.

https://gitlab.gnome.org/GNOME/yelp/issues/123
---
 src/yelp-application.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/src/yelp-application.c b/src/yelp-application.c
index 47a69991..c7fd646c 100644
--- a/src/yelp-application.c
+++ b/src/yelp-application.c
@@ -370,12 +370,27 @@ YelpApplication *
 yelp_application_new (void)
 {
     YelpApplication *app;
+    char *app_id = NULL;
+    char *yelp = NULL;
+
+    if (g_file_test ("/.flatpak-info", G_FILE_TEST_EXISTS)) {
+        GKeyFile *kf = g_key_file_new ();
+        g_key_file_load_from_file (kf, "/.flatpak-info", G_KEY_FILE_NONE, NULL);
+        app_id = g_key_file_get_string (kf, "Application", "name", NULL);
+        yelp = g_strconcat (app_id, ".Help", NULL);
+        g_key_file_unref (kf);
+    }
+    else {
+        yelp = g_strdup ("org.gnome.Yelp");
+    }
 
     app = g_object_new (YELP_TYPE_APPLICATION,
-                        "application-id", "org.gnome.Yelp",
+                        "application-id", yelp,
                         "flags", G_APPLICATION_HANDLES_COMMAND_LINE,
                         "inactivity-timeout", 5000,
                         NULL);
+    g_free (app_id);
+    g_free (yelp);
 
     return app;
 }
-- 
2.12.0

